noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_tail.nr

```diff
@@ -46,6 +47,7 @@ impl PrivateKernelTailCircuitPrivateInputs {
         assert_eq(
             array_length(previous_public_inputs.end.public_call_stack), 0, "Public call stack must be empty when executing the tail circuit"
         );
+        assert(is_empty(previous_public_inputs.public_teardown_call_request) == true, "Public teardown call request must be empty when executing the tail circuit");
```

Comment:
For the sake of my own understanding: why are these two assertions here? Doesn't the private kernel tail run at the very end of the private phase? Wouldn't the public calls requests be in here? So why do we assert they're empty?